docs/modules/ROOT/pages/index.adoc

```diff
@@ -43,10 +43,236 @@ Once the BOM is added, simply add the `io.quarkiverse.artemis:quarkus-artemis-jm
+            ...
+    ) {
+        ...
+        this.serverLocator = serverLocator;
+        ...
+    }
+}
+----
+
+.Injection of default `ConnectionFactory` by its explicit name when using `quarkus-artemis-jms`
+[source,java,subs=attributes+]
+----
+public class MyClass {
+    ...
+    private final ConnectionFactory connectionFactory;
+    ...
+
+    public MyClass (
+            ...
+            @Identifier("<default>") ConnectionFactory connectionFactory,
+            ...
+    ) {
+        ...
+        this.connectionFactory = connectionFactory;
+        ...
+    }
+}
+----
+
+Named configurations, on the other hand, are not configured as `@Default` bean and must therefore be injected by name. The name of the bean is the name of the configuration, and `@Identifier(...)` is used to qualify the name. So for example, if we define a configuration "named-1":
+
+.Named configuration "named-1"
+[source,properties,subs=attributes+]
+----
+quarkus.artemis."named-1".url=...
+quarkus.artemis."named-1".username=...
+quarkus.artemis."named-1".password=...
+----
+
+we can inject it with:
+
+.Injection of named `ServerLocator` "named-1" when using extension `quarkus-artemis-core`
+[source,java,subs=attributes+]
+----
+public class MyClass {
+    ...
+    private final ServerLocator serverLocator;
+    ...
+
+    public MyClass (
+            ...
+            @Identifier("named-1") ServerLocator serverLocator,
+            ...
+    ) {
+        ...
+        this.serverLocator = serverLocator;
+        ...
+    }
+}
+----
+
+.Injection of named `ConnectionFactory` "named-1" when using extension `quarkus-artemis-jms`
+[source,java,subs=attributes+]
 ----
+public class MyClass {
+    ...
+    private final ConnectionFactory connectionFactory;
+    ...
+
+    public MyClass (
+            ...
+            @Identifier("named-1") ConnectionFactory connectionFactory,
+            ...
+    ) {
+        ...
+        this.connectionFactory = connectionFactory;
+        ...
+    }
+}
+----
+
+IMPORTANT: We strongly recommend the usage of link:https://javadoc.io/doc/io.smallrye.common/smallrye-common-annotation/latest/io/smallrye/common/annotation/Identifier.html[`@Identifier` annotation] instead of the link:https://www.javadoc.io/doc/javax/javaee-api/latest/javax/inject/Named.html[`@Named` annotation]. Some other feature rely on the usage of `@Identfier`. Please see section <<_health_checks>> for details.
+
+== Configuration detection at build-time
+We took special care so the configurations behave "as intuitively as possible". This means that if no connection-related configuration (`enabled`, `url`, `useranme`, `password`, `devservices...`, `health-exclude`, `xa-enabled`) is present at build-time, it is assumed that the  configuration is not used and disabled. Therefore, if we want to use any configuration, but not configure it, we should set `quarkus.artemis.enabled` / `quarkus.artemis."named-configuration".enabled` to *true* to explicitly enable the configuration.
```

Comment:
I understand what is being said here, but I don't understand the why. What would be the use case to leave out all configurations, but do enable the config 😕

Reply:
Camel. At least that's how I interpreted a corner-case of ["*Camel has some alternative ways to configure Artemis, e.g. end users can produce their own ConnectionFactory.*"](https://github.com/quarkusio/quarkus/issues/18231).

Reply:
Then I would change it like:

```
We took special care so the configurations behave "as intuitively as possible". In general this means that if no connection-related configuration fields (`enabled`, `url`, `username`, `password`, `devservices...`, `health-exclude`, `xa-enabled`) are present at build-time it is disabled. If you produce your own ConnectionFactory you can set `enabled` to `true` to explicitly enable the configuration.```

But I don't understand when you need this, because as far as I understand now, is that they even produce their own default ConnectionFactory. So they don't need to configure one via our extension because `url` is optional now.

Reply:
Through `@Identtifier`, they get helath checks for free.

Reply:
Ah yes.

> We took special care so the configurations behave "as intuitively as possible". In general this means that if no connection-related configuration fields (enabled, url, username, password, devservices..., health-exclude, xa-enabled) are present at build-time it is disabled. If you produce your own ConnectionFactory you can set enabled to true to explicitly enable the configuration to include it in health-checks.

Like this?

Reply:
Or your own wording, but adding the example or use case makes it understandable.

Reply:
>  ...
> > ...If you produce your own ConnectionFactory you can set `enabled` to `true` to explicitly enable the configuration.
>  ...

This is actually not correct. Our extension does nothing with beans provided in code (no wrapping, nothing). The only "service" we provide, as stated above, is inclusion in health checks when the bean is annotated with `@Identifier(...)`.

Reply:
But how do they use `enabled then?

Reply:
Why should they? The `ConnectionFactory` is provided in-code. We cannot enalbe/disalbe this bean since its creation is out of our control. Or am I misunderstanding something?

Reply:
FTR: https://quarkusio.zulipchat.com/#narrow/pm-with/285062,403386-pm/near/306979669